src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -519,6 +519,204 @@ public static async ValueTask<FormulaValue> FilterTable(EvalVisitor runner, Eval
             var rows = await LazyFilterAsync(runner, context, arg0.Rows, arg1).ConfigureAwait(false);
 
             return new InMemoryTableValue(irContext, rows);
+        }
+
+        // Join(t1, t2, LeftRecord.Id = RightRecord.RefId, JoinType.Inner)
+        public static async ValueTask<FormulaValue> JoinTables(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            var leftTable = (TableValue)args[0];
+            var rightTable = (TableValue)args[1];
+            var predicate = (LambdaFormulaValue)args[2];
+            var joinType = (OptionSetValue)args[3];
+            var leftRenaming = (RecordValue)args[4];
+            var rightRenaming = (RecordValue)args[5];

```

Comment:
comment above only shows 4 args; what do these last 2 mean?

Reply:
Is this right:
Source may have arbitrary number of args. 
And then IR transform will effectively transform it and "group" them into 2 args (leftRenaming, RightRenaming) ?